app/controllers/tickets_controller.rb

```diff
@@ -19,4 +19,21 @@ def create
 
     redirect_to root_path
   end
+
+  before_action :authenticate_user!
+  def buy
+    ticket = Ticket.find(params[:ticket_id])
+    ticket.buy_ticket(ticket.id, current_user.id)
```

Comment:
Tutaj bardzo fajnie widać, że ma Pan intuicję jak robić ekstrakcję metod do innych klas. Większosć grup tworzy całą logikę w kontrolerze, co nie jest poprawne. Ale poszedłbym o krok dalej i zmienił to w następujący sposób:

1) Programujemy obiektowo, więc staramy się przekazywać całe obiekty a nie proste typy jak `id`
czyli zamiast `ticket.buy_ticket(ticket.id, current_user.id)`
zrobiłbym `ticket.buy_ticket(ticket, current_user)`

2) Metoda `buy_ticket` wywoływana jest już na instancji `ticket` więc nie potrzeba jej przekazywać kolejny raz samej siebie (pierwszy argument u Pana to `ticket.id`)

wystarczy
`ticket.buy_ticket(current_user)`
a w models/ticket.rb:
```
def buy_ticket(user)
  self.bought_by = user # zadziała też samo "bought_by = user", self nie jest potrzebne
end
```
3) Nazwa metody `buy_ticket` sugeruje, że metoda obsługuje już całą sprzedaż. Jednak ona przypisuje jedynie kupującego do ticketu. Starałbym się używać bardziej opisowego języka w nazwach metod.

4) Wszystko co tutaj jesy wyciągnąłbym do interaktora, tak żeby kontroler wyglądał w ten sposób:

```ruby
def buy
  ticket = Ticket.find(params[:ticket_id])

  outcome = BuyTicket.run(ticket: ticket, user: current_user)
  if outcome.valid?
    flash[:notice] = 'Bilet zostal kupiony'
  else
    flash[:notice] = @ticket.errors.full_messages.join('. ') 
  end
  
  redirect_to root_path
end
```

Przykład interaktora można znaleźć tutaj https://github.com/piotrimbierowicz/test_app_uwr/tree/master/app/interactions
i tutaj https://github.com/AaronLasseigne/active_interaction